components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -390,13 +391,24 @@ public User getUser(String userId) throws CharonException {
                 }
                 return Collections.emptyList();
             } else {
+                //get claims related to SCIM claim dialect
+                claims = carbonClaimManager.getAllClaimMappings(SCIMCommonConstants.SCIM_CLAIM_DIALECT);
+                List<String> claimURIList = new ArrayList<>();
+                for (ClaimMapping claim : claims) {
```

Comment:
You don't have to go through this for loop if user didn't enable SCIMProviderConstants.ELEMENT_NAME_SHOW_ALL_USER_DETAILS right? Shall we move this logic to inside the if condition "if (Boolean.parseBoolean(IdentityUtil.getProperty(SCIMProviderConstants.ELEMENT_NAME_SHOW_ALL_USER_DETAILS)))"